test/e2e/memtierd.test-suite/damon.source.sh

```diff
@@ -1,67 +1,87 @@
-damon-idlepage-setup() {
-    if [ "$distro" != "debian-sid" ]; then
-        error "damon-idlepage-setup is implemented only for distro=debian-sid"
+#!/bin/bash
+# usage: kernel-download $version
+kernel-download() {
+    vm-command "[ -d linux ]" && vm-command "rm -rf linux"
+
+    vm-command "apt-get update" || error "failed to refresh apt package DB"
+    vm-install-pkg wget git-core
+
+    local version=${1:-$linux}
+    if [[ ! -z ${version} ]]; then
+        url="https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/snapshot/linux-${version}.tar.gz"
+        vm-command "wget -q -c ${url} -O - | tar -xz" || error "failed to download kernel"
+        vm-command "mv linux-${version} linux"
+    else
+        vm-command "git clone https://github.com/torvalds/linux"
+    fi
+}
+
+# usage: kernel-patch $version $patch_file
+kernel-patch() {
+    # download linux kernel source code
+    if [[ ! -z $1 ]] || [[ ! -z $linux ]]; then
+        kernel-download $1 || error "failed to download kernel"
```

Comment:
How about doing only patching in this function. If the user has not downloaded already the kernel version of her choice (that is, if there is not "linux" directory on VM), then this function can error out.

If this function only applies given patch file to kernel sources on vm, then a user could call this function several times in order to apply multiple patches.

We can assume the user call (for maximal flexibility) for instance
```
kernel-download 6.6.2
kernel-patch my.patch
kernel-patch my.hack
kernel-build-and-install
vm-reboot
```

even here I'm not fully sure if we should separate build and install to separate steps in user perspective...  but definitely user might want to do still something after successful kernel install and before reboot, so let's separate those.